src/platform/nrfconnect/FactoryDataProvider.h

```diff
@@ -40,11 +40,43 @@ struct InternalFlashFactoryData
         return CHIP_NO_ERROR;
     }
 
+#ifdef CONFIG_FPROTECT
```

Comment:
I wonder.. if it wouldn't be better to have a dedicated config for Matter and Factory Data (e.g. `CHIP_FACTORY_DATA_WRITE_PROTECT`). Someone may want to enable FPROTECT for other purpose etc.

Reply:
This would also hide nicely the limitation we have. If someone would like to implement different memory map (like that settings are not close to the factory data) then he can disable `CHIP_FACTORY_DATA_WRITE_PROTECT` but still use factory data module.

Reply:
For the second comment, this is already possible, but as you said, currently the user needs to disable the whole FPROTECT support in such a case, so adding a dedicated Konfig makes sense. Thanks for the hint.